### ISSUE TITLE
Regression in Ansible 2.2.1.0, with Jinja (2.8.1)

### ISSUE THREAD

**Issue by CorbanR** [ansible/ansible#20956]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
jinja2, template

##### ANSIBLE VERSION
```
ansible 2.2.1.0
  config file =
  configured module search path = Default w/o overrides
```

##### OS / ENVIRONMENT
Running Ansible from both OSX Sierra and from a Ubuntu Trusty (control server)
Jinja2 `version: 2.8.1`

##### SUMMARY
Possible regression in Ansible `2.2.1.0`, and Jinja.

In Ansible `2.2.0.0` using the template module with Ansible variables was working as expected. After upgrading to `2.2.1.0` extra character is showing up in template.

Seems related to https://github.com/ansible/ansible/issues/20568. Creating a different issue due to context.

##### STEPS TO REPRODUCE
Folder Layout:
test_playbook/inventory
test_playbook/group_vars/all
test_playbook/playbook.yml
test_playbook/templates

inventory
```
[jump]
jump.domain.net
```

playbook
```
---
- name: jump test
  hosts: jump
  gather_facts: yes
  tasks:
    - name: debug
      debug:
        msg: "{{ trusted_proxies }}"

    - name: Copy jinja template
      template:
        src: test.j2
        dest: /home/user/test
```

group_vars/all
```
---
trusted_proxies: "['{{ ansible_eth0.ipv4.address }}']"
```

templates/test.j2
```
'{{ trusted_proxies }}'
```

command run from root
`ansible-playbook -i inventory playbook.yml -vv`
##### EXPECTED RESULTS
contents of `/home/user/test`  should be
```
'['10.0.14.1']'
```
##### ACTUAL RESULTS
contents of `/home/user/test` when using ansible `2.2.1.0`
```
'[u'10.0.14.1]'
```

In both versions of Ansible the debug msg was the same (no added characters)
```
TASK [debug] *******************************************************************
task path: /private/tmp/test_playbook/playbook.yml:6
ok: [jump.domain.net] => {
    "msg": [
        "10.0.14.1"
    ]
}
```

**Comment by jctanner** [comment]
@CorbanR what is going to consume the result of the template operation? Is it supposed to be json compliant? Would a |to_json filter suffice?

needs_info

**Comment by CorbanR** [comment]
@jctanner initially this caused me an issue in a gitlab playbook

The config line that caused me problems was `gitlab_rails['trusted_proxies'] = {{ gitlab_trusted_proxies }}` settings in gitlab.rb.. 

It's odd that the behavior from `2.2.0.0` to `2.2.1.0` has changed in this instance 

the `|to_json` did in fact fix the issue in `2.2.1.0` :) Thank you.

**Comment by CorbanR** [comment]
I guess ill close the issue. Just seemed odd that the behavior changed.
